app/src/androidTest/java/org/mozilla/tv/firefox/ui/robots/NavigationOverlayRobot.kt

```diff
@@ -80,6 +84,10 @@ class NavigationOverlayRobot {
 
     fun assertActivityFinishing(activity: MainActivityTestRule) = assertTrue(activity.activity.isFinishing)
 
+    fun waitForYouTubeRedirects(url: String) {
+        device.wait(Until.findObject(By.textContains(url)), 5000)
```

Comment:
This function is very generic, but the name is extremely specific.  I think this should be broken into two methods, `waitUntilYouTubeHomeLoads` and `waitUntil YouTubeVideoLoads`.  Better that the robot contains that kind of implementation detail.